lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1089 @@
+      LoadDataFromPDB(PDBFilePath.str());
+    } else {
+      // Usualy the PDB files will be generated on a different machine,
+      // so the location read from the debug directory wont be up to date. At
+      // first try to find it in the current `.` dir.
+      auto PDBFileNameOnly = PDBFilePath.substr(PDBFilePath.find_last_of('\\')
+                                                + 1);
+      if (llvm::sys::fs::exists(PDBFileNameOnly.str()))
+        LoadDataFromPDB(PDBFileNameOnly.str());
+    }
+  }
+
+  if (not ThePDBFile) {
+    revng_log(DILogger, "Unable to find PDB file.");
+    return;
+  } else {
+    // According to llvm/docs/PDB/PdbStream.rst, the `Signature` was never
+    // used the way as it was the initial idea. Instead, GUID is a 128-bit
+    // identifier guaranteed to be unique ID for both executable and
+    // corresponding PDB.
+    codeview::GUID GUIDFromExe;
+    llvm::copy(DebugInfo->PDB70.Signature, std::begin(GUIDFromExe.Guid));
+    auto PDBInfoStrm = ThePDBFile->getPDBInfoStream();
+    if (!PDBInfoStrm)
+      revng_log(DILogger, "No PDB Info stream found.");
+    else {
+      codeview::GUID GUIDFromPDBFIle = PDBInfoStrm->getGuid();
+      if (GUIDFromExe != GUIDFromPDBFIle)
+        revng_log(DILogger, "Signatures from exe and PDB file mismatch.");
+    }
+  }
+
+  PDBImporterImpl ModelCreator(*this);
+  ModelCreator.run(*TheNativeSession);
+}
+
+// ==== Implementation of PDBImporterTypeVisitor ==== //
+
+Error PDBImporterTypeVisitor::visitTypeBegin(CVType &Record) {
+  return visitTypeBegin(Record, TypeIndex::fromArrayIndex(Types.size()));
+}
+
+Error PDBImporterTypeVisitor::visitTypeBegin(CVType &Record, TypeIndex TI) {
+  CurrentTypeIndex = TI;
+  return Error::success();
+}
+
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               FieldListRecord &FieldList) {
+  if (auto EC = visitMemberRecordStream(FieldList.Data, *this))
+    return EC;
+  return Error::success();
+}
+
+// Determine the pointer size based on CodeView/PDB data.
+static uint32_t getPointerSizeInBytes(codeview::PointerKind K) {
+  switch (K) {
+  case codeview::PointerKind::Near64:
+    return 8;
+  case codeview::PointerKind::Near32:
+    return 4;
+  default:
+    // TODO: Handle all pointer kinds.
+    return 4;
+  }
+}
+
+// Parse LF_POINTER.
+Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
+                                               PointerRecord &Ptr) {
+  using namespace model;
+  auto TypedefTy = makeType<TypedefType>();
+  auto TypeDef = cast<TypedefType>(TypedefTy.get());
+
+  TypeIndex RefTy = Ptr.getReferentType();
+
+  if (ProcessedTypes.count(RefTy) == 0 and RefTy.isSimple())
+    createPrimitiveType(RefTy);
+
+  if (ProcessedTypes.count(RefTy)) {
+    auto PtrSize = getPointerSizeInBytes(Ptr.getPointerKind());
+    std::vector<Qualifier> Qualifiers{ Qualifier::createPointer(PtrSize) };
+    QualifiedType Qty(ProcessedTypes[RefTy], Qualifiers);
```

Comment:
Let's use meaningful, not-abbreviated names. The codebase looks less cryptic this way